clio-transfer-model/src/main/scala/org/broadinstitute/clio/transfer/model/TransferMetadata.scala

```diff
@@ -32,19 +32,23 @@ trait TransferMetadata[M <: TransferMetadata[M]] { self: M =>
 
   /**
     * Return a copy of this object in which all files in `pathsToMove`
-    * have been moved into a destination directory.
+    * have been moved to the given destination.
     *
-    * Fails if `destination` doesn't end with '/'.
+    * If `pathsToMove` contains > 1 element, `destination` must end with '/'.
     */
-  def moveAllInto(destination: String): M
-
-  /**
-    * Returns copy of this object in which a singular file-path has
-    * been sent to a new destination.
-    *
-    * Fails if `pathsToMove` contains more than one element.
-    */
-  def setSinglePath(destination: String): M
+  def moveInto(destination: String): M = {
+    val pathMapper =
+      if (destination.endsWith("/")) { (opt: Option[String]) =>
+        opt.map(path => moveIntoDirectory(path, destination))
+      } else if (pathsToMove.length > 1) {
+        sys.error(
```

Comment:
Is there a logger that we should be writing to here?

Reply:
Nope, this will raise a `RuntimeException` which will get caught by the client logic, rolled up and logged at the top level.

Reply:
I think in general logging should be done at the (root) catch point, not the throw point, because only the caller knows what needs to be done with an exception.